### PR TITLE
kotlin-language-server: init at 0.7.0

### PR DESCRIPTION
**Pull request by enderger**
###### Motivation for this change
Currently, there is no package for the Kotlin Lanugage Server, often used by IDE plugins for the development of Kotlin.
I'm a bit new to git, and accidentally closed the origional PR

###### Things done

- 

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)

- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:2616]: Please put this into a separate commit with the message ``maintainers: add enderger``.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 105110` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>kotlin-language-server</li>
  </ul>
</details>

**Comment by enderger** [comment]
For some reason, Git created a new branch with the same name for the new commits.

**Comment by lukegb** [comment]
A few comments:

* (tiny nit) 2-space indentation is more common in nixpkgs :)
* For open-source packages we prefer to have them built from source rather than using binary releases; #17342 tracks making this sort of thing easier in nixpkgs. #72306 is an example of doing it "by hand", which is another option.
* The ./bin/kotlin-language-server doesn't run "alone" (it expects java to be on the PATH already) which means your `buildInputs` isn't really going to help - it isn't actually used in building your package. You should probably *remove* the buildInputs line, add "makeWrapper" to nativeBuildInputs, and add to your installPhase:
```
    wrapProgram $out/bin/kotlin-language-server \
      --prefix PATH : ${jre}/bin
```

Some Git advice for splitting: my advice here is probably to carefully recommit things. If you're using the CLI, the best way (in my opinion, other people will have different advice) is to:
```
$ git reset --soft HEAD^
# You've now removed the last commit, but the changes are still in the index ready to commit, so commit the maintainer-list first:
$ git commit maintainers/maintainer-list.nix
# ...write a commit message for the maintainer-list.nix file
$ git commit
# ...write a commit message for the rest of the files
$ git push --force-with-lease
# and you're done!
```

**Comment by enderger** [comment]
I've fixed the indentation and the wrapping issue, however I was unable to get Gradle (I tried the included wrapper and the Nix package) to build the package.

**Review by lukegb — APPROVED** [review]

**Comment by enderger** [comment]
@SuperSandro2000 I also split the commits using **lukegb**'s suggestion.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 105110` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>kotlin-language-server</li>
  </ul>
</details>

File: pkgs/top-level/all-packages.nix
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:9570]: This does not belong into compilers. Please put it under software development tools or similar.

Comment (enderger) [reply pkgs/top-level/all-packages.nix:9570]: Um, this is the ‘kotlin-language-server’ package, not the ‘kotlin’ package.

**Comment by SuperSandro2000** [comment]
@enderger Please fix the merge conflict.

**Comment by enderger** [comment]
I've resolved it.

**Comment by SuperSandro2000** [comment]
Please do not merge master into PRs. Please resolve that cleanly.

**Comment by enderger** [comment]
Hey, sorry for dropping this. I was quite naive when it comes to Git when I made this PR. I may try redoing this package at some point, but I'll close the PR since tit is hopelessly out-of-date.

**Pull request closed without merging**
